src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/ref/Microsoft.Extensions.DependencyInjection.Abstractions.cs

```diff
@@ -49,10 +59,19 @@ public partial interface IServiceScopeFactory
     {
         Microsoft.Extensions.DependencyInjection.IServiceScope CreateScope();
     }
+    public partial interface ISupportKeyedService
+    {
+        object? GetKeyedService(System.Type serviceType, object serviceKey);
+        object GetRequiredKeyedService(System.Type serviceType, object serviceKey);
+    }
     public partial interface ISupportRequiredService
     {
         object GetRequiredService(System.Type serviceType);
     }
+    public static partial class KeyedService
+    {
+        public static object AnyKey { get { throw null; } }
```

Comment:
This could be `ServiceDescriptor.AnyServiceKey` or similar instead of adding a new class. I don't have any good reason for why it should be either way. Someone may argue that this class should be named `ServiceKey`